apps/avifdec.c

```diff
@@ -210,52 +235,73 @@ int main(int argc, char * argv[])
+        goto cleanup;
+    }
+
+    result = avifDecoderNthImage(decoder, frameIndex);
+    if (result != AVIF_RESULT_OK) {
+        fprintf(stderr, "ERROR: Failed to decode image: %s\n", avifResultToString(result));
+        returnCode = 1;
+        goto cleanup;
+    }
+
+    printf("Image decoded: %s\n", inputFilename);
+    printf("Image details:\n");
+    avifImageDump(decoder->image, 0, 0, decoder->progressiveState);
+
+    if (ignoreICC && (decoder->image->icc.size > 0)) {
+        printf("[--ignore-icc] Discarding ICC profile.\n");
+        avifImageSetProfileICC(decoder->image, NULL, 0);
+    }
 
-        avifAppFileFormat outputFormat = avifGuessFileFormat(outputFilename);
-        if (outputFormat == AVIF_APP_FILE_FORMAT_UNKNOWN) {
-            fprintf(stderr, "Cannot determine output file extension: %s\n", outputFilename);
+    avifAppFileFormat outputFormat = avifGuessFileFormat(outputFilename);
+    if (outputFormat == AVIF_APP_FILE_FORMAT_UNKNOWN) {
+        fprintf(stderr, "Cannot determine output file extension: %s\n", outputFilename);
+        returnCode = 1;
+    } else if (outputFormat == AVIF_APP_FILE_FORMAT_Y4M) {
+        if (!y4mWrite(outputFilename, decoder->image)) {
             returnCode = 1;
-        } else if (outputFormat == AVIF_APP_FILE_FORMAT_Y4M) {
-            if (!y4mWrite(outputFilename, avif)) {
-                returnCode = 1;
-            }
-        } else if (outputFormat == AVIF_APP_FILE_FORMAT_JPEG) {
-            // Bypass alpha multiply step during conversion
-            if (rawColor) {
-                avif->alphaPremultiplied = AVIF_TRUE;
-            }
-            if (!avifJPEGWrite(outputFilename, avif, jpegQuality, chromaUpsampling)) {
-                returnCode = 1;
-            }
-        } else if (outputFormat == AVIF_APP_FILE_FORMAT_PNG) {
-            if (!avifPNGWrite(outputFilename, avif, requestedDepth, chromaUpsampling)) {
-                returnCode = 1;
-            }
-        } else {
-            fprintf(stderr, "Unrecognized file extension: %s\n", outputFilename);
+        }
+    } else if (outputFormat == AVIF_APP_FILE_FORMAT_JPEG) {
+        // Bypass alpha multiply step during conversion
+        if (rawColor) {
+            decoder->image->alphaPremultiplied = AVIF_TRUE;
+        }
+        if (!avifJPEGWrite(outputFilename, decoder->image, jpegQuality, chromaUpsampling)) {
+            returnCode = 1;
+        }
+    } else if (outputFormat == AVIF_APP_FILE_FORMAT_PNG) {
+        if (!avifPNGWrite(outputFilename, decoder->image, requestedDepth, chromaUpsampling)) {
             returnCode = 1;
         }
     } else {
-        printf("ERROR: Failed to decode image: %s\n", avifResultToString(decodeResult));
-        avifDumpDiagnostics(&decoder->diag);
+        fprintf(stderr, "Unrecognized file extension: %s\n", outputFilename);
         returnCode = 1;
     }
-    avifDecoderDestroy(decoder);
-    avifImageDestroy(avif);
+
+cleanup:
+    if (decoder) {
```

Comment:
This null pointer check for `decoder` is not necessary because `decoder` cannot be null. See line 238:

`avifDecoder * decoder = avifDecoderCreate();`